RapidReact/src/main/java/frc/robot/subsystem/ClimberSubsystem.java

```diff
@@ -1,112 +1,311 @@
+        {
+          nextState = ClimbState.Idle;
+          idleInit();
+        }
+        else
+        {
+          nextState = ClimbState.Retract;      
+        }
+        break;
+    }
+
+    currentClimbState = nextState;
+  }
+
 
   @Override
   public void disable() {
-    climber.set(0);
+    climber1.set(0);
+    climber2.set(0);
   }
 
   public void toggleClimberEnabled() { // uses 2 PS button
-    climberEnabled = !climberEnabled;
-    
-    if (climberEnabled){
+    enabledClimber = !enabledClimber;
+
+    if (enabledClimber) {
       climbState.log(LogLevel.GENERAL, "climberEnabled");
-    } else{
+    } else {
       climbState.log(LogLevel.GENERAL, "climberDisabled");
     }
   }
 
+  public void manualElevatorExtend() { //uses up button
+    if (autoClimb) return;
 
+    // TODO PERSISTNET: LIMIT SWITCHES https://docs.ctre-phoenix.com/en/stable/ch13_MC.html#limit-switches
+    // TODO PERSISTENT: you should have the joystick/ button move the motion magic setpoint, not the motor in PWM mode
 
-  public void fixedHookToggler() { //uses R1 button
-    if (config.enablePneumatics) {
-      fixedHookToggleState = !fixedHookToggleState;
-      if (!fixedHookToggleState) {
-        fixedHookSolenoid.set(Value.kForward);
-        hookToggleState.log(LogLevel.GENERAL, "fixedHookTime");
-      } else {
-        fixedHookSolenoid.set(Value.kReverse);
-        hookToggleState.log(LogLevel.GENERAL, "fixedHookTime");
-      }
-    }
-  }
-
-  public void elevatorExtend() { //uses up button
-    climber.set(ControlMode.PercentOutput, climbOutput.currentValue());
+    climber1.set(ControlMode.PercentOutput, climbOutput.currentValue());
+    climber2.set(ControlMode.PercentOutput, climbOutput.currentValue());
+    climberMotorPosition.log(climber1.getSelectedSensorPosition());
 
     climbState.log(LogLevel.GENERAL, "elevatorExtend");
   }
 
-  public void elevatorRetract() { //uses down button
-    climber.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+  public void manualElevatorRetract() { //uses down button
+    if (autoClimb) return;
+
+    // TODO PERSISTNET: LIMIT SWITCHES https://docs.ctre-phoenix.com/en/stable/ch13_MC.html#limit-switches
+    // TODO PERSISTENT: you should have the joystick/ button move the motion magic setpoint, not the motor in PWM mode
+
+    climber1.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+    climber2.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+    climberMotorPosition.log(climber1.getSelectedSensorPosition());
 
     climbState.log(LogLevel.GENERAL, "elevatorRetract");
   }
 
   public void elevatorStop() {
-    climber.set(0);
+    climber1.set(0);
+    climber2.set(0);
     climbState.log(LogLevel.GENERAL, "climbStopped");
   }
 
-  public void climbAuto() { //uses TPAD button
+  // autoClimb() both enables autoclimb and sets the flag to true
+  // autoClimbReleased() just sets the flag to false
+  public void autoClimb() { //uses TPAD button
```

Comment:
Make sure that the state machine is in the right state when autoClimb is called. Should you add the line "currentClimbState = ClimbState.Idle;" that's run if the climber isn't in the autoclimb mode already? Your stopAutoClimb function may already take care of this.

Reply:
This should be taken care of by line 151 I believe